### DISCUSSION TITLE
Filter values in drop down

### DISCUSSION THREAD

**Discussion by shortsam1** [serenity-is/Serenity#5634] [Help] [answered]
Can you filter the values in a drop down list so as the list only displays 2 or 3 certain entries out of the whole list?  I was trying to use the filtervalue option as i could not see anything else that would do it.

Out of the list of 13 entries that exist in the list, for this pop up I only need to return values with ids equal to 1,2 and 4.

This example returned no results.

` [LookupEditor(typeof(Core.Entities.HpRow), FilterValue =1)
  public Int32 HpId { get; set; `

**Comment by erehtolleh** [comment]
You have to specify a FilterField also which is the field name you are filtering the value on.

**Reply by shortsam1** [reply]
Thanks

**Comment by shortsam1** [comment, marked as answer]
I ended up creating an editor as follows:

```
namespace OnlineForms.Report {

    @Serenity.Decorators.registerEditor()
    export class HpEditor extends
        Serenity.LookupEditorBase<Serenity.LookupEditorOptions, Core.HpRow> {

        constructor(container: JQuery, opt: Serenity.LookupEditorOptions) {
            super(container, opt);
        }

        protected getLookupKey() {
            return Core.HpRow.lookupKey;
        }

        protected getItems(lookup: Q.Lookup<Core.HpRow>) {
            return super.getItems(lookup).filter(x =>
                x.HpId === 1 || x.HpId === 2 || x.HpId === 4);
        }
    }
}
```
